tests/mitmproxy_addons/callback.py

```diff
@@ -20,10 +20,25 @@
 #   response_body: { some json object },
 #   response_code: 200,
 # }
-# Currently this is a read-only callback. The response cannot be modified, but side-effects can be
-# taken. For example, tests may wish to terminate a client prior to the delivery of a response but
-# after the server has processed the request, or the test may wish to use the response as a
-# synchronisation point for a Waiter.
+# The response to this request can control what gets returned to the client. The response object:
+# {
+#    "respond_status_code": 200,
+#    "respond_body": { "some": "json_object" }
+# }
+# If {} is sent back, the response is not modified. Likewise, if `respond_body` is set but
+# `respond_status_code` is not, only the response body is modified, not the status code, and vice versa.
```

Comment:
This only applies to `callback_response_url`, right?

I think it might be clearer to document the request and response for the two URLs completely separately, even if there's a bit of duplication.

Reply:
Yes.

Sounds good, I'm going to document this in the README where it's more easily accessible. But that will be in the next PR.